pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/realtime/PinotLLCRealtimeSegmentManager.java

```diff
@@ -515,7 +519,8 @@ private void createNewSegmentZKMetadata(TableConfig tableConfig, PartitionLevelS
     newSegmentZKMetadata.setTableName(realtimeTableName);
     newSegmentZKMetadata.setSegmentName(segmentName);
     newSegmentZKMetadata.setCreationTime(creationTimeMs);
-    newSegmentZKMetadata.setStartOffset(startOffset);
+    // TODO Issue 5953 Use string offset in the metadata
+    newSegmentZKMetadata.setStartOffset(Long.parseLong(startOffset.toString()));
```

Comment:
will we introduce new methods for setting offset in the metadata? What happens to old segments?

Reply:
I was looking at metadata yesterday, and they are stored as strings. Luckily, the metadata classes don't construct objects directly from json. So, there may be no changes needed for metadata, only change the access method.

Coming soon to a PR near you.